# SOME DESCRIPTIVE TITLE
# Copyright (C) YEAR Free Software Foundation, Inc.
# This file is distributed under the same license as the PACKAGE package.
# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
#
msgid ""
msgstr ""
"Project-Id-Version: git-docs-ja 0.0\n"
"POT-Creation-Date: 2021-11-04 05:35+0900\n"
"PO-Revision-Date: 2022-05-24 10:52+0900\n"
"Last-Translator: kuma35\n"
"Language-Team: Japanese\n"
"Language: ja_JP\n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=UTF-8\n"
"Content-Transfer-Encoding: 8bit\n"

#. type: Title =
#: ../Documentation-sedout/technical/api-error-handling.txt:2
#, no-wrap
msgid "Error reporting in git"
msgstr "Error reporting in git"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:6
msgid ""
"`BUG`, `die`, `usage`, `error`, and `warning` report errors of various kinds."
msgstr "`BUG`, `die`, `usage`, `error`, `warning` は、さまざまな種類のエラーを報告します。"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:9
msgid ""
"`BUG` is for failed internal assertions that should never happen, i.e. a bug "
"in git itself."
msgstr "`BUG` は、決して起こらないはずの失敗した内部アサーション、つまりgit自体のバグのためのものです。"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:12
msgid ""
"`die` is for fatal application errors.  It prints a message to the user and "
"exits with status 128."
msgstr "`die` は致命的なアプリケーションエラー用です。 ユーザーにメッセージを出力し、ステータス128で終了します。"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:16
msgid ""
"`usage` is for errors in command line usage.  After printing its message, it "
"exits with status 129.  (See also `usage_with_options` in the link:api-parse-"
"options.html[parse-options API].)"
msgstr "`usage` は、コマンドラインの使用法のエラー用です。 メッセージを出力した後、ステータス129で終了します(link:api-parse-options.html[parse-options API] の `usage_with_options` も参照してください)。"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:20
msgid ""
"`error` is for non-fatal library errors.  It prints a message to the user "
"and returns -1 for convenience in signaling the error to the caller."
msgstr "`error` は、致命的ではないライブラリエラー用です。 ユーザーにメッセージを出力し、呼び出し元にエラーを通知するのに便利なように -1 を返します。"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:25
msgid ""
"`warning` is for reporting situations that probably should not occur but "
"which the user (and Git) can continue to work around without running into "
"too many problems.  Like `error`, it returns -1 after reporting the "
"situation to the caller."
msgstr "`warning` は、おそらく発生しないはずの状況を報告するためのものですが、ユーザー(およびGit)はあまり多くの問題に遭遇することなく回避を続けることができます。 `error` と同様に、呼び出し元に状況を報告した後、 -1 を返します。"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:28
msgid ""
"These reports will be logged via the trace2 facility. See the \"error\" "
"event in link:api-trace2.txt[trace2 API]."
msgstr "これらのレポートは、trace2機能を介してログに記録されます。 link:api-trace2.txt[trace2 API] の「error」イベントを参照してください。"

#. type: Title -
#: ../Documentation-sedout/technical/api-error-handling.txt:30
#, no-wrap
msgid "Customizable error handlers"
msgstr "Customizable error handlers"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:37
msgid ""
"The default behavior of `die` and `error` is to write a message to stderr "
"and then exit or return as appropriate.  This behavior can be overridden "
"using `set_die_routine` and `set_error_routine`.  For example, \"git daemon"
"\" uses set_die_routine to write the reason `die` was called to syslog "
"before exiting."
msgstr "`die` と `error` のデフォルトの動作は、stderrにメッセージを書き込んでから、必要に応じて終了(exit)または戻ることです。 この振る舞いは、 `set_die_routine` と `set_error_routine` を使用してオーバーライドできます。 例えば、 `git daemon` は set_die_routine を使って、 `die` が呼ばれた理由を syslog に書き込んでから終了するようにしています。"

#. type: Title -
#: ../Documentation-sedout/technical/api-error-handling.txt:39
#, no-wrap
msgid "Library errors"
msgstr "Library errors"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:43
msgid ""
"Functions return a negative integer on error.  Details beyond that vary from "
"function to function:"
msgstr "各関数はエラー時に負の整数を返します。それ以外の詳細は機能ごとに異なります:"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:47
msgid ""
"Some functions return -1 for all errors.  Others return a more specific "
"value depending on how the caller might want to react to the error."
msgstr "一部の関数は、すべてのエラーに対して -1 を返します。 それ以外の関数は、呼び出し元がエラーにどのように反応したいかに応じて、より具体的な値を返します。"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:50
msgid ""
"Some functions report the error to stderr with `error`, while others leave "
"that for the caller to do."
msgstr "一部の関数はエラーを `error` でstderrに報告しますが、それ以外の関数はそれを呼び出し元に任せます。"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:53
msgid ""
"errno is not meaningful on return from most functions (except for thin "
"wrappers for system calls)."
msgstr "errno は、ほとんどの関数から戻ったときに意味がありません(システムコールの薄いラッパー(thin wrappers)を除く)。"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:55
msgid "Check the function's API documentation to be sure."
msgstr "各関数のAPIドキュメントを確認してください。"

#. type: Title -
#: ../Documentation-sedout/technical/api-error-handling.txt:57
#, no-wrap
msgid "Caller-handled errors"
msgstr "Caller-handled errors"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:63
msgid ""
"An increasing number of functions take a parameter 'struct strbuf *err'.  On "
"error, such functions append a message about what went wrong to the 'err' "
"strbuf.  The message is meant to be complete enough to be passed to `die` or "
"`error` as-is.  For example:"
msgstr "ますます多くの関数がパラメータ `struct strbuf *err` を取るようになっています。エラーの場合、そのような関数は、 `err` strbufに何がうまくいかなかったかについてのメッセージを追加します。 メッセージは、そのまま `die` または `error` に渡されるのに十分なほど完全であることを意図しています。 例えば:"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:66
#, no-wrap
msgid ""
"\tif (ref_transaction_commit(transaction, &err))\n"
"\t\tdie(\"%s\", err.buf);\n"
msgstr ""
"\tif (ref_transaction_commit(transaction, &err))\n"
"\t\tdie(\"%s\", err.buf);\n"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:69
msgid ""
"The 'err' parameter will be untouched if no error occurred, so multiple "
"function calls can be chained:"
msgstr "エラーが発生しなかった場合、 `err` パラメーターは変更されないため、複数の関数呼び出しを連鎖させることができます:"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:75
#, no-wrap
msgid ""
"\tt = ref_transaction_begin(&err);\n"
"\tif (!t ||\n"
"\t    ref_transaction_update(t, \"HEAD\", ..., &err) ||\n"
"\t    ret_transaction_commit(t, &err))\n"
"\t\tdie(\"%s\", err.buf);\n"
msgstr ""
"\tt = ref_transaction_begin(&err);\n"
"\tif (!t ||\n"
"\t    ref_transaction_update(t, \"HEAD\", ..., &err) ||\n"
"\t    ret_transaction_commit(t, &err))\n"
"\t\tdie(\"%s\", err.buf);\n"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:78
msgid ""
"The 'err' parameter must be a pointer to a valid strbuf.  To silence a "
"message, pass a strbuf that is explicitly ignored:"
msgstr "`err` パラメーターは、有効なstrbufへのポインターでなければなりません。 メッセージを黙らせるには、明示的に無視されるstrbufを渡します:"

#. type: Plain text
#: ../Documentation-sedout/technical/api-error-handling.txt:81
#, no-wrap
msgid ""
"\tif (thing_that_can_fail_in_an_ignorable_way(..., &err))\n"
"\t\t/* This failure is okay. */\n"
"\t\tstrbuf_reset(&err);\n"
msgstr ""
"\tif (thing_that_can_fail_in_an_ignorable_way(..., &err))\n"
"\t\t/* This failure is okay. */\n"
"\t\tstrbuf_reset(&err);\n"
